ui/design/src/components/Avatar/styled-avatar.ts

```diff
@@ -64,6 +67,21 @@ const StyledAvatar = styled.div<AvatarProps>`
         return;
     }
   }}
+  ${props => {
+    const { borderColor } = props;
+    switch (borderColor) {
```

Comment:
since you are returning `undefined` in the default case, it would be nicer to just pass the property name of the color and check if it's valid. If it's valid add a border with the specified color, otherwise return undefined